pkgs/http/lib/src/client.dart

```diff
@@ -210,6 +210,17 @@ Client? get zoneClient {
 ///  runWithClient(() => runApp(const MyApp()), clientFactory);
 /// }
 /// ```
+///
+/// If [runWithClient] is used and the environment defines
+/// `no_default_http_client=true` then generated binaries may be smaller e.g.
+/// ```shell
+/// $ flutter build appbundle --dart-define=no_default_http_client=true ...
+/// $ dart compile exe --define=no_default_http_client=true ...
+/// ```
+///
+/// If `no_default_http_client=true` is set then any call to the [Client]
+/// factory (i.e. `Client()`) outside of the [Zone] created by [runWithClient]
+/// will throw [StateError].
```

Comment:
[optional] I think you can link directly to the constructor in DartDoc with [Client.new].

```suggestion
/// If `no_default_http_client=true` is set then any call to [Client.new]
/// outside of the [Zone] created by [runWithClient] will throw [StateError].
```